src/Authentication/GraphPhpLeagueAccessTokenProvider.php

```diff
@@ -61,6 +64,27 @@ public function __construct(
             $tokenBaseServiceUrl,
             $nationalCloud
         );
-        parent::__construct($tokenRequestContext, $scopes, $allowedHosts, $oauthProvider);
+        $offlineAccessEnabled = in_array('offline_access', array_map(fn ($scope) => trim(strtolower($scope)), $scopes), true);
+        if (!$offlineAccessEnabled) {
+            $scopes[] = 'offline_access';
```

Comment:
I'm not sure what's the rationale behind adding that scope forcibly?

Reply:
Thinking it's safe to always request a refresh token? This covers the case that developers may not be aware of the `offline_access` scope being required.

MSAL actually adds a bunch of other [defaults](https://github.com/AzureAD/microsoft-authentication-library-for-java/blob/bc6ab89c9caec96f312ed2e1d7dc0637a166ea74/msal4j-sdk/src/main/java/com/microsoft/aad/msal4j/OAuthAuthorizationGrant.java#L25)

Reply:
I'm not sure all flows support returning a refresh token, I know that implicit flow didn't and the service would barf at you back then.

Reply:
Thanks for context. Reverted this.